File: totaleconomy-core/src/main/java/com/ericgrandt/totaleconomy/data/entity/CurrencyEntity.java
package com.ericgrandt.totaleconomy.data.entity;

import java.math.BigDecimal;
import java.time.Instant;

public record CurrencyEntity(
    int id,
    String code,
    String name,
    String pluralName,
    String symbol,
    int fractionalDigits,
    BigDecimal startingBalance,
    boolean isDefault,
    Instant createdAt
) {
}

File: totaleconomy-core/src/main/kotlin/com/ericgrandt/totaleconomy/data/entity/CurrencyEntity.kt
package com.ericgrandt.totaleconomy.data.entity

import kotlin.time.Instant

data class CurrencyEntity(
    val id: Int,
    val code: String,
    val name: String,
    val pluralName: String,
    val symbol: String?,
    val fractionalDigits: Int,
    val isDefault: Boolean,
    val createdAt: Instant,
)
